### PR TITLE
Fix deleting product when default variant is deleted

### PR DESCRIPTION
**Pull request by IKarbowiak**
Don't delete product when the product default variant is deleted.

[SALEOR-1190](https://app.clickup.com/t/2549495/SALEOR-1190)

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1190 Deleting default variant should not delete product](https://app.clickup.com/t/2549495/SALEOR-1190)

File: saleor/graphql/product/bulk_mutations/products.py [hunk 1]
```diff
@@ -371,11 +371,25 @@ def perform_mutation(cls, _root, info, ids, **data):
...
 
         # delete order lines for deleted variants
         order_models.OrderLine.objects.filter(pk__in=order_line_pks).delete()
 
+        # set new product default varaint if any has been removed
+        products = models.Product.objects.filter(
+            pk__in=product_pks, default_variant__isnull=True
+        )
+        for product in products:
+            product.default_variant = product.variants.first()
```
Comment (patrys) [inline saleor/graphql/product/bulk_mutations/products.py:390]: Shouldn't this only be done if we're removing the current default?

Comment (patrys) [reply saleor/graphql/product/bulk_mutations/products.py:390]: Nevermind, I see it's only done for products missing a default.

File: saleor/graphql/product/bulk_mutations/products.py [hunk 2]
```diff
@@ -371,11 +371,25 @@ def perform_mutation(cls, _root, info, ids, **data):
...
+            .distinct()
+            .values_list("pk", flat=True)
+        )
+
         response = super().perform_mutation(_root, info, ids, **data)
 
         # delete order lines for deleted variants
         order_models.OrderLine.objects.filter(pk__in=order_line_pks).delete()
 
+        # set new product default varaint if any has been removed
```
Comment (maarcingebala) [inline saleor/graphql/product/bulk_mutations/products.py:385]: You have been visited by the grammar police 👮  
```suggestion
        # set new product default variant if any has been removed
```

**Review by maarcingebala — APPROVED** [review]

File: saleor/graphql/product/bulk_mutations/products.py [hunk 3]
```diff
@@ -371,11 +371,25 @@ def perform_mutation(cls, _root, info, ids, **data):
             ).values_list("pk", flat=True)
         )
 
+        product_pks = list(
```
Comment (fowczarek) [inline saleor/graphql/product/bulk_mutations/products.py:374]: I wonder if all this logic from `perform_mutation` should be done atomically?

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/product/bulk_mutations/products.py (hunk 3 above)
Comment (IKarbowiak) [reply saleor/graphql/product/bulk_mutations/products.py:374]: 👍  I'll add a transaction there.

**Review by fowczarek — APPROVED** [review]

**Pull request merged**